runtime/js/src/main/kotlin/kotlinx/serialization/DynamicObjectParser.kt

```diff
@@ -56,6 +60,18 @@ class DynamicObjectParser(): AbstractSerialFormat() {
             }
         }
 
+        override fun decodeTaggedLong(tag: String): Long {
+            val obj = getByTag(tag)
+            val number = obj as? Double ?: throw SerializationException("$obj is not a Number")
+            val canBeConverted = number.isFinite() && floor(number) == number
```

Comment:
FYI there are built-in [`Number.isInteger`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger) and [Number.isSafeInteger](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger)
Just thought you might find them useful =)
Current implementation has the benefit of being supported on IE though... So your choice =)

Reply:
Yes, I've looked them up in docs before writing this. I wanted to distinguish cases 'not a number' from 'not an integer' from 'not a safe int' without a lot of `js(...)` code blocks – all these methods are not exposed from `kotlin.Number`.

Reply:
Ok)